### ISSUE TITLE
Unable to refer column from a table with Primary Key of the same table

### ISSUE THREAD

**Issue by WarienX** [RobinBuschmann/sequelize-typescript#519]
@RobinBuschmann, We need to set a Foreign Key reference with the Primary Key of the same table. But, we haven't been able to achieve this in Sequelize TypeScript.
For example 
![image 1](https://user-images.githubusercontent.com/26925053/51536606-2d805180-1e72-11e9-962e-75bed38bb42b.png)

Any suggestions would be greatly appreciated..

**Comment by RobinBuschmann** [comment]
Hey @WarienX, can you show me how you attempts look like? (Maybe this helps already: The `@ForeeignKey` annotation is only a helper for the `@BelongsTo`, `@HasMany` and so on annotation. Which means,  `@ForeeignKey` doesn't work without them)

**Comment by WarienX** [comment]
@RobinBuschmann, Thanks for replying but, we were able to get it working. It was not referencing because both the column names had different data types. After we changed the data type it worked. Thanks once again. But we currently need your assistance on another problem regarding composite primary keys. Any Help would be greatly appreciated. The link is added below..

https://github.com/RobinBuschmann/sequelize-typescript/issues/523
